### PR TITLE
Notebook refactor

### PR DESCRIPTION
**Pull request by nikhilmandlik**
Resolves: https://github.com/nasa/openmct/issues/2825
* [x] Address comments in https://github.com/nasa/openmct/pull/2755/files
* [x] Add unit tests

### CODE REVIEW

**Comment by akhenry** [comment]
@nikhilmandlik This is pretty close. There are just those two small issues outstanding

File: src/plugins/notebook/components/notebook-embed.vue [hunk 1]
```diff
@@ -77,95 +80,45 @@ export default {
...
-                        };
-                    } else {
-                        console.log('You cancelled the annotation!!!');
-                    }
-
-                    done(true);
-                }
-            }).show(this.embed.snapshot.src);
+            painterroInstance = new PainterroInstance();
+            painterroInstance.callback = this.updateSnapshot;
```
[the remark this answers is not in the document]
Comment (nikhilmandlik) [reply src/plugins/notebook/components/notebook-embed.vue]: Updated

**Review by akhenry — DISMISSED** [review]

**Comment by nikhilmandlik** [comment]
resolved conflicts

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Review by akhenry — APPROVED** [review]

**Pull request merged**